pkg/generated/restapi/configure_rekor_server.go

```diff
@@ -174,21 +178,81 @@ func setupMiddlewares(handler http.Handler) http.Handler {
 	return handler
 }
 
+type httpRequestFields struct {
+	requestMethod string
+	requestURL    string
+	requestSize   int64
+	status        int
+	responseSize  int
+	userAgent     string
+	remoteIP      string
+	latency       time.Duration
+	protocol      string
+}
+
+func (h *httpRequestFields) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddString("requestMethod", h.requestMethod)
+	enc.AddString("requestUrl", h.requestURL)
+	enc.AddString("requestSize", fmt.Sprintf("%d", h.requestSize))
+	enc.AddInt("status", h.status)
+	enc.AddString("responseSize", fmt.Sprintf("%d", h.responseSize))
+	enc.AddString("userAgent", h.userAgent)
+	enc.AddString("remoteIP", h.remoteIP)
+	enc.AddString("latency", fmt.Sprintf("%.9fs", h.latency.Seconds())) // formatted per GCP expectations
+	enc.AddString("protocol", h.protocol)
+	return nil
+}
+
 // We need this type to act as an adapter between zap and the middleware request logger.
-type logAdapter struct {
+type zapLogEntry struct {
+	r *http.Request
 }
 
-func (l *logAdapter) Print(v ...interface{}) {
-	log.Logger.Info(v...)
+func (z *zapLogEntry) Write(status, bytes int, header http.Header, elapsed time.Duration, extra interface{}) {
+	var fields []interface{}
+
+	// follows https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry as a convention
+	// append HTTP Request / Response Information
+	scheme := "http"
+	if z.r.TLS != nil {
+		scheme = "https"
+	}
+	httpRequestObj := &httpRequestFields{
+		requestMethod: z.r.Method,
+		requestURL:    fmt.Sprintf("%s://%s%s", scheme, z.r.Host, z.r.RequestURI),
+		requestSize:   z.r.ContentLength,
+		status:        status,
+		responseSize:  bytes,
+		userAgent:     z.r.Header.Get("User-Agent"),
+		remoteIP:      z.r.RemoteAddr,
```

Comment:
Any reason to log IP? Some PII risk of being able to correlate requests across artifacts, gather usage data, etc